[CI] HistoryIntegrationTests testPayloadInputWithDotsInFieldNameWorks failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/e3qdnup7y5say/tests/:x-pack:plugin:watcher:internalClusterTest/org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests/testPayloadInputWithDotsInFieldNameWorks


**Reproduction line:**
```
./gradlew ":x-pack:plugin:watcher:internalClusterTest" --tests "org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests.testPayloadInputWithDotsInFieldNameWorks" -Dtests.seed=F934C0DCCD1128EC -Dtests.locale=und -Dtests.timezone=Pacific/Kanton -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests#testPayloadInputWithDotsInFieldNameWorks`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testPayloadInputWithDotsInFieldNameWorks'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed

  at __randomizedtesting.SeedInfo.seed([F934C0DCCD1128EC:A5592375180E9950]:0)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:724)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:416)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:756)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:509)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.performPhaseOnShard(AbstractSearchAsyncAction.java:314)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.run(AbstractSearchAsyncAction.java:250)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:466)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.start(AbstractSearchAsyncAction.java:225)
  at org.elasticsearch.action.search.TransportSearchAction.executeSearch(TransportSearchAction.java:1213)
  at org.elasticsearch.action.search.TransportSearchAction.executeLocalSearch(TransportSearchAction.java:961)
  at org.elasticsearch.action.search.TransportSearchAction.lambda$executeRequest$5(TransportSearchAction.java:379)
  at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:245)
  at org.elasticsearch.index.query.Rewriteable.rewriteAndFetch(Rewriteable.java:109)
  at org.elasticsearch.index.query.Rewriteable.rewriteAndFetch(Rewriteable.java:77)
  at org.elasticsearch.action.search.TransportSearchAction.executeRequest(TransportSearchAction.java:491)
  at org.elasticsearch.action.search.TransportSearchAction.doExecute(TransportSearchAction.java:335)
  at org.elasticsearch.action.search.TransportSearchAction.doExecute(TransportSearchAction.java:115)
  at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:96)
  at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:68)
  at org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:196)
  at org.elasticsearch.client.internal.node.NodeClient.executeLocally(NodeClient.java:105)
  at org.elasticsearch.client.internal.node.NodeClient.doExecute(NodeClient.java:83)
  at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:138)
  at org.elasticsearch.client.internal.FilterClient.doExecute(FilterClient.java:55)
  at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:138)
  at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:124)
  at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:36)
  at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:43)
  at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:16)
  at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertResponse(ElasticsearchAssertions.java:379)
  at org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount(ElasticsearchAssertions.java:301)
  at org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests.lambda$testPayloadInputWithDotsInFieldNameWorks$2(HistoryIntegrationTests.java:154)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1384)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1369)
  at org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests.testPayloadInputWithDotsInFieldNameWorks(HistoryIntegrationTests.java:152)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

  Caused by: org.elasticsearch.action.NoShardAvailableActionException: (No message provided)


```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -85,9 +85,6 @@ tests:
 - class: org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests
   method: testCacheFileCreatedAsSparseFile
   issue: https://github.com/elastic/elasticsearch/issues/110801
-- class: "org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests"
-  issue: "https://github.com/elastic/elasticsearch/issues/110885"
-  method: "testPayloadInputWithDotsInFieldNameWorks"
 - class: org.elasticsearch.nativeaccess.PreallocateTests
   method: testPreallocate
   issue: https://github.com/elastic/elasticsearch/issues/110948

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/HistoryIntegrationTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/test/integration/HistoryIntegrationTests.java
@@ -7,6 +7,7 @@
 package org.elasticsearch.xpack.watcher.test.integration;
 
 import org.elasticsearch.action.admin.indices.mapping.get.GetMappingsResponse;
+import org.elasticsearch.action.search.SearchRequestBuilder;
 import org.elasticsearch.protocol.xpack.watcher.PutWatchResponse;
 import org.elasticsearch.search.SearchHit;
 import org.elasticsearch.search.sort.SortBuilders;
@@ -15,6 +16,7 @@ import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xcontent.XContentType;
 import org.elasticsearch.xpack.core.watcher.actions.ActionStatus;
 import org.elasticsearch.xpack.core.watcher.client.WatchSourceBuilder;
+import org.elasticsearch.xpack.core.watcher.history.HistoryStoreField;
 import org.elasticsearch.xpack.core.watcher.input.Input;
 import org.elasticsearch.xpack.core.watcher.support.xcontent.XContentSource;
 import org.elasticsearch.xpack.core.watcher.transport.actions.execute.ExecuteWatchRequestBuilder;
@@ -71,10 +73,7 @@ public class HistoryIntegrationTests extends AbstractWatcherIntegrationTestCase 
 
         new ExecuteWatchRequestBuilder(client()).setId("test_watch").setRecordExecution(true).get();
 
-        assertBusy(() -> {
-            flushAndRefresh(".watcher-history-*");
-            assertHitCount(prepareSearch(".watcher-history-*"), 1);
-        });
+        assertBusy(() -> { assertHitCount(getWatchHistory(), 1); });
     }
 
     // See https://github.com/elastic/x-plugins/issues/2913
@@ -104,10 +103,7 @@ public class HistoryIntegrationTests extends AbstractWatcherIntegrationTestCase 
 
         new ExecuteWatchRequestBuilder(client()).setId("test_watch").setRecordExecution(true).get();
 
-        assertBusy(() -> {
-            refresh(".watcher-history*");
-            assertHitCount(prepareSearch(".watcher-history*").setSize(0), 1);
-        });
+        assertBusy(() -> { assertHitCount(getWatchHistory(), 1); });
 
         // as fields with dots are allowed in 5.0 again, the mapping must be checked in addition
         GetMappingsResponse response = indicesAdmin().prepareGetMappings(".watcher-history*").get();
@@ -149,10 +145,7 @@ public class HistoryIntegrationTests extends AbstractWatcherIntegrationTestCase 
 
         new ExecuteWatchRequestBuilder(client()).setId("test_watch").setRecordExecution(true).get();
 
-        assertBusy(() -> {
-            refresh(".watcher-history*");
-            assertHitCount(prepareSearch(".watcher-history*").setSize(0), 1);
-        });
+        assertBusy(() -> { assertHitCount(getWatchHistory(), 1); });
 
         // as fields with dots are allowed in 5.0 again, the mapping must be checked in addition
         GetMappingsResponse response = indicesAdmin().prepareGetMappings(".watcher-history*").get();
@@ -186,8 +179,7 @@ public class HistoryIntegrationTests extends AbstractWatcherIntegrationTestCase 
         WatchStatus status = new GetWatchRequestBuilder(client()).setId("test_watch").get().getStatus();
 
         assertBusy(() -> {
-            refresh(".watcher-history*");
-            assertResponse(prepareSearch(".watcher-history*").setSize(1), searchResponse -> {
+            assertResponse(getWatchHistory(), searchResponse -> {
                 assertHitCount(searchResponse, 1);
                 SearchHit hit = searchResponse.getHits().getAt(0);
 
@@ -231,6 +223,16 @@ public class HistoryIntegrationTests extends AbstractWatcherIntegrationTestCase 
                 is(nullValue())
             );
         });
+    }
+
+    /*
+     * Returns a SearchRequestBuilder containing up to the default number of watch history records (10) if the .watcher-history* is ready.
+     * Otherwise it throws an AssertionError.
+     */
+    private SearchRequestBuilder getWatchHistory() {
+        ensureGreen(HistoryStoreField.DATA_STREAM);
+        flushAndRefresh(".watcher-history-*");
+        return prepareSearch(".watcher-history-*");
     }
 
 }

